gallery_dl/extractor/steamgriddb.py

```diff
@@ -0,0 +1,211 @@
+                urls.append(asset["fake_png"])
+
+            asset["count"] = len(urls)
+            yield Message.Directory, asset
+            for asset["num"], url in enumerate(urls, 1):
+                yield Message.Url, url, text.nameext_from_url(url, asset)
+
+    def _call(self, endpoint: str, **kwargs):
+        data = self.request(self.root + endpoint, **kwargs).json()
+        if not data["success"]:
+            raise exception.StopExtraction(data["error"])
+        return data["data"]
+
+
+class SteamgriddbAssetsExtractor(SteamgriddbExtractor):
+    """Base class for extracting a list of assets"""
+
+    def __init__(self, match):
+        SteamgriddbExtractor.__init__(self, match)
+        list_type = match.group(1)
+        id = int(match.group(2))
+        self.game_id = id if list_type == "game" else None
+        self.collection_id = id if list_type == "collection" else None
+        self.page = int(match.group(3) or 1)
+
+    def assets(self):
+        limit = 48
+        page = min(self.page - 1, 0)
+
+        sort = self.config("sort", "score_desc")
+        if sort not in ("score_desc", "score_asc", "score_old_desc",
+                        "score_old_asc", "age_desc", "age_asc"):
+            raise exception.StopExtractor("Invalid sort '%s'", sort)
+
+        json = {
+            "static"  : self.config("static", True),
+            "animated": self.config("animated", True),
+            "humor"   : self.config("humor", True),
+            "nsfw"    : self.config("nsfw", True),
+            "epilepsy": self.config("epilepsy", True),
+            "untagged": self.config("untagged", True),
+
+            "asset_type": self.asset_type,
+            "limit": limit,
+            "order": sort,
+        }
+        if self.valid_dimensions:
+            json["dimensions"] = self.config_list(
+                "dimensions", "dimension", self.valid_dimensions)
+        json["styles"] = self.config_list("styles", "style", self.valid_styles)
+        json["languages"] = self.config_list(
+            "languages", "language", LANGUAGE_CODES)
+        file_types = self.config_list(
+            "file-types", "file type", self.valid_file_types)
+        json["mime"] = [FILE_EXT_TO_MIME[i] for i in file_types]
+
+        if self.game_id:
+            json["game_id"] = [self.game_id]
+        else:
+            json["collection_id"] = self.collection_id
+
+        while True:
+            json["page"] = page
+
+            data = self._call(
+                "/api/public/search/assets", method="POST", json=json)
+            for asset in data["assets"]:
+                if not asset.get("game"):
+                    asset["game"] = data["game"]
+                yield asset
+
+            if data["total"] > limit * page:
+                page += 1
+            else:
+                break
+
+    def config_list(self, key, type_name, valid_values):
+        value = self.config(key, ["all"])
```

Comment:
I really dislike having a non-trivial value as default value, `["all"]` in this case. Leave it out and check for None instead.

```suggestion
        value = self.config(key)
        if value is None:
            return ["all"]
```

or maybe even
```
        if value is None or value == "all":
```